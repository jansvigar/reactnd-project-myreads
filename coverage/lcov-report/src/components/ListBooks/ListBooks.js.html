<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/ListBooks/ListBooks.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/ListBooks</a> ListBooks.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/6</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import * as BooksAPI from '../../BooksAPI';
import BookShelf from '../BookShelf/BookShelf';
import * as Constants from '../../constants'
import { camelCase } from '../../utils'
import './ListBooks.css';
&nbsp;
class ListBooks extends Component {
  state = {
    books: []
  };
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentDidMount() {
<span class="cstat-no" title="statement not covered" >    this.updateShelvedBooksList();</span>
  };
&nbsp;
  updateShelvedBooksList = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >    BooksAPI.getAll().then(<span class="fstat-no" title="function not covered" >(b</span>ooks) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >      this.setState({ books });</span>
    });
  }
&nbsp;
  render() {
    return (
      &lt;div className="list-books"&gt;
          &lt;div className="list-books-title"&gt;
            &lt;h1&gt;MyReads&lt;/h1&gt;
          &lt;/div&gt;
          &lt;div className="list-books-content"&gt;
            &lt;div&gt;
              { Constants.SHELVES.map( shelf =&gt; (
                &lt;BookShelf key={ shelf }
                           title={ shelf }
                           books={ this.state.books.filter(<span class="fstat-no" title="function not covered" >bo</span>ok =&gt;
<span class="cstat-no" title="statement not covered" >                             book.shelf === camelCase(shelf))</span>
                           }
                           updateShelvedBooksList={ this.updateShelvedBooksList }
                /&gt;
              )) }
            &lt;/div&gt;
          &lt;/div&gt;
          &lt;div className="open-search"&gt;
            &lt;Link to="/search"&gt;Add a book&lt;/Link&gt;
          &lt;/div&gt;
        &lt;/div&gt;
      );
  };
}
&nbsp;
export default ListBooks;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Aug 11 2017 00:33:58 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
