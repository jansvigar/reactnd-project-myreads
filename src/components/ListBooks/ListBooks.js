import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import * as BooksAPI from '../../BooksAPI';
import BookShelf from '../BookShelf/BookShelf';
import * as Constants from '../../constants'
import { camelCase } from '../../utils'
import './ListBooks.css';

class ListBooks extends Component {
  state = {
    books: []
  };

  componentDidMount() {
    this.updateShelvedBooksList();
  };

  updateShelvedBooksList = () => {
    BooksAPI.getAll().then((books) => {
      this.setState({ books });
    });
  }

  render() {
    return (
      <div className="list-books">
          <div className="list-books-title">
            <h1>MyReads</h1>
          </div>
          <div className="list-books-content">
            <div>
              { Constants.SHELVES.map( shelf => (
                <BookShelf key={ shelf }
                           title={ shelf }
                           books={ this.state.books.filter(book =>
                             book.shelf === camelCase(shelf))
                           }
                           updateShelvedBooksList={ this.updateShelvedBooksList }
                />
              )) }
            </div>
          </div>
          <div className="open-search">
            <Link to="/search">Add a book</Link>
          </div>
        </div>
      );
  };
}

export default ListBooks;
