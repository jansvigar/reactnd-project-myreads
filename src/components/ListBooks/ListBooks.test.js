import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ListBooks from './ListBooks';

describe('<ListBooks />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ListBooks />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
