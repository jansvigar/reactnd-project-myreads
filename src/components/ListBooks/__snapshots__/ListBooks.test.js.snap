// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListBooks /> should render correctly 1`] = `
<div
  className="list-books"
>
  <div
    className="list-books-title"
  >
    <h1>
      MyReads
    </h1>
  </div>
  <div
    className="list-books-content"
  >
    <div>
      <BookShelf
        books={Array []}
        title="Currently Reading"
        updateShelvedBooksList={[Function]}
      />
      <BookShelf
        books={Array []}
        title="Want to Read"
        updateShelvedBooksList={[Function]}
      />
      <BookShelf
        books={Array []}
        title="Read"
        updateShelvedBooksList={[Function]}
      />
    </div>
  </div>
  <div
    className="open-search"
  >
    <Link
      replace={false}
      to="/search"
    >
      Add a book
    </Link>
  </div>
</div>
`;
